src/plugins/kibana_utils/common/url/encode_uri_query.ts

```diff
@@ -43,15 +43,38 @@ export function encodeUriQuery(val: string, pctEncodeSpaces = false) {
 
 export const encodeQuery = (
   query: ParsedQuery,
-  encodeFunction: (val: string, pctEncodeSpaces?: boolean) => string = encodeUriQuery
-) =>
-  transform(query, (result: any, value, key) => {
+  encodeFunction: (val: string, pctEncodeSpaces?: boolean) => string = encodeUriQuery,
+  pctEncodeSpaces = true
+): ParsedQuery =>
+  transform<any, ParsedQuery>(query, (result, value, key) => {
     if (key) {
       const singleValue = Array.isArray(value) ? value.join(',') : value;
 
       result[key] = encodeFunction(
         singleValue === undefined || singleValue === null ? '' : singleValue,
-        true
+        pctEncodeSpaces
       );
     }
   });
+
+/**
+ * Method to help modify url query params.
+ *
+ * @param params
+ * @param key
+ * @param value
+ */
+export const modifyParams = (params: string, key: string, value?: string) => {
+  const queryParams = parse(params);
+
+  if (value) {
```

Comment:
While currently `value` is defined to be a string or undefined in TS, I think that checking explicitly if value is `undefined` is more correct. (for example if a boolean false value is passed in at some point)

Reply:
Yes, I think that is more foolproof. See 5c51a678f13bbd1934ca37efcfc19f336fecd4c3